tests/dataflow/test_unified_systolic.py

```diff
@@ -6,112 +6,151 @@
 import allo.dataflow as df
 import allo.backend.hls as hls
 import numpy as np
+import pytest
 
 M, N, K = 4, 4, 4
 Rt, Ct = 2, 2
-# Rt, Ct = 4, 4
-# M, N, K = 16, 16, 16
-# Rt, Ct = 4, 4
-# Rt, Ct = 16, 16
 P0, P1 = Rt + 2, Ct + 2
 
 
 @df.region()
 def top():
-    fifo_R = df.array(df.pipe(dtype=int32, shape=(), depth=4), shape=(P0, P1))
-    fifo_C = df.array(df.pipe(dtype=int32, shape=(), depth=4), shape=(P0, P1))
+    # interconnect
+    fifo_R = df.array(df.pipe(dtype=int32, shape=(), depth=16), shape=(P0, P1 - 1))
+    fifo_C = df.array(df.pipe(dtype=int32, shape=(), depth=16), shape=(P0 - 1, P1))
+    inst_broad = df.array(df.pipe(dtype=bool, shape=(), depth=4), shape=(P1 - 1,))
+    inst_chain = df.array(df.pipe(dtype=bool, shape=(), depth=4), shape=(P0 - 1, P1))
 
     @df.kernel(mapping=[P0, P1])
-    def gemm(A: int32[M, K], B: int32[K, N], flowtag: bool, C: int32[M, N]):
+    def gemm(A: int32[M, K], B: int32[K, N], inst: bool[1], C: int32[M, N]):
```

Comment:
Any reasons changing back to `bool[1]`?

Reply:
Just wanna show some decoding later. Like extracting the first elements in instructions as the configuration signal. I can change it back to the original single bool arguments if you think it's better.

Reply:
Then how about creating a test case calculating two consecutive GEMM, one using output-stationary and the other using weight-stationary

Reply:
I think this might be a higher dimensional test, and we could add that in later PR. I'll let it just be a single bool argument here.

Reply:
It depends on where you'd like to make the decoding happen. We can talk about that in details offline later.